[GTK][WPE] Remove webkit_web_view_new_with_related_view from new API too

carlosgcampos opened the pull request:
#### 57a26786ef09798e808a70a8e6e2da499326b8e2
<pre>
[GTK][WPE] Remove webkit_web_view_new_with_related_view from new API too
<a href="https://bugs.webkit.org/show_bug.cgi?id=251426">https://bugs.webkit.org/show_bug.cgi?id=251426</a>

Reviewed by Adrian Perez de Castro.

It&apos;s confusing because it&apos;s not equivalent to just calling
g_object_new() with releated-view property set.

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestPrinting.cpp:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::createWebView):

Canonical link: <a href="https://commits.webkit.org/259885@main">https://commits.webkit.org/259885@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4cdee52097171b9249ad4c39c37360af36672f70

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106303 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15357 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39142 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115496 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175600 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110213 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16799 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6573 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98519 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115183 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112067 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12782 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95767 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/40335 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94657 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27410 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82025 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8597 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28762 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9107 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5331 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14720 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48308 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10643 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3682 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (changes requested):
OK, you convinced me this is a good idea, but I'm still going to request changes here:

 * Add a critical in `constructed` if the object is constructed with `related-view` and also `web-context` or `network-session`
 * What about `is-controlled-by-automation`?
 * Please update the migration guide; it currently says this constructor is the only one that was not removed

carlosgcampos committed:
[GTK][WPE] Remove webkit_web_view_new_with_related_view from new API too
https://bugs.webkit.org/show_bug.cgi?id=251426

Reviewed by Adrian Perez de Castro.

It's confusing because it's not equivalent to just calling
g_object_new() with releated-view property set.

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestPrinting.cpp:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::createWebView):

Canonical link: https://commits.webkit.org/259885@main
